File: Euclid/Network/Streams/Request.cs
using Euclid.Util.Extensions;
using System;
using System.Collections.Generic;

namespace Euclid.Network.Streams
{
    public class Request
    {

        /// <summary>
        /// Get the message header
        /// </summary>
        public string Header
        {
            get; private set;
        }

        /// <summary>
        /// Get the message buffer
        /// </summary>
        public string Content
        {
            get; private set;
        }

        /// <summary>
        /// Get the message body with characters replaced
        /// </summary>
        public string MessageBody
        {
            get
            {
                string consoleText = Content;

                for (int i = 0; i < 14; i++)
                {
                    consoleText = consoleText.Replace(Convert.ToString((char)i), "[" + i + "]");
                }

                return consoleText;
            }
        }

        #region Constructors

        public Request(string header, string request)
        {
            this.Header = header;
            this.Content = request;
        }

        #endregion

        #region Methods

        /// <summary>
        /// Skip some of the content
        /// </summary>
        /// <param name="v"></param>
        public void Skip(int num)
        {
            if (Content.Length < num)
                return;

            Content = Content.Substring(num);
        }

        /// <summary>
        /// Packet handle to get count of arguments by delimeter
        /// </summary>
        public int GetArgumentAmount(string delimeter = " ")
        {
            if (this.Content.Length == 0) return 0;
            return this.Content.Split(delimeter).Length;
        }

        /// <summary>
        /// Packet handle to get specific argument
        /// </summary>
        public string GetArgument(int index, string delimeter = " ")
        {
            if (index < 0)
                index = 0;

            if (GetArgumentAmount(delimeter) < 1)
            {
                return null;
            }

            return this.Content.Split(delimeter)[index];
        }

        /// <summary>
        /// Get key/values separated by '=' 
        /// </summary>
        public Dictionary<string, string> GetKeyValues(string content = null)
        {
            var values = new Dictionary<string, string>();

            if (content == null)
                content = this.Content;

            foreach (var line in content.Split((char)13))
            {
                if (line.Trim().Length < 1)
                {
                    continue;
                }

                if (!line.Contains('='))
                {
                    continue;
                }

                int offset = line.IndexOf('=');
                values[line.Substring(0, offset)] = line.Substring(offset + 1).FilterInput();
            }

            return values;
        }

        #endregion

        /// <summary>
        /// Dispose handler of request
        /// </summary>
        public void Dispose()
        {

        }
    }
}

File: euclid/src/main/java/euclid/network/streams/Request.java
package euclid.network.streams;

import euclid.util.extensions.StringExtensions;

import java.util.LinkedHashMap;
import java.util.Map;

public class Request {

    private String header;
    private String content;

    public Request(String header, String content) {
        this.header = header;
        this.content = content;
    }

    public String getHeader() {
        return header;
    }

    public String getContent() {
        return content;
    }

    public String getMessageBody() {
        String consoleText = content;
        for (int i = 0; i < 14; i++) {
            consoleText = consoleText.replace(String.valueOf((char) i), "[" + i + "]");
        }
        return consoleText;
    }

    public void skip(int num) {
        if (content.length() < num) {
            return;
        }
        content = content.substring(num);
    }

    public int getArgumentAmount(String delimeter) {
        if (content.length() == 0) {
            return 0;
        }
        return splitLiteral(content, delimeter).length;
    }

    public int getArgumentAmount() {
        return getArgumentAmount(" ");
    }

    public String getArgument(int index, String delimeter) {
        if (index < 0) {
            index = 0;
        }
        if (getArgumentAmount(delimeter) < 1) {
            return null;
        }
        return splitLiteral(content, delimeter)[index];
    }

    public String getArgument(int index) {
        return getArgument(index, " ");
    }

    public Map<String, String> getKeyValues(String content) {
        Map<String, String> values = new LinkedHashMap<>();
        if (content == null) {
            content = this.content;
        }

        String[] lines = content.split(String.valueOf((char) 13), -1);
        for (String line : lines) {
            if (line.trim().length() < 1) {
                continue;
            }
            if (!line.contains("=")) {
                continue;
            }
            int offset = line.indexOf('=');
            values.put(line.substring(0, offset), StringExtensions.filterInput(line.substring(offset + 1)));
        }
        return values;
    }

    public Map<String, String> getKeyValues() {
        return getKeyValues(null);
    }

    private static String[] splitLiteral(String content, String delimiter) {
        if (content.isEmpty()) {
            return new String[0];
        }
        java.util.List<String> parts = new java.util.ArrayList<>();
        int start = 0;
        int idx;
        while ((idx = content.indexOf(delimiter, start)) != -1) {
            parts.add(content.substring(start, idx));
            start = idx + delimiter.length();
        }
        parts.add(content.substring(start));
        return parts.toArray(new String[0]);
    }
}
